### ISSUE TITLE
Unicode characters cause terminal output to split

### ISSUE THREAD

**Issue by TheQuinbox** [nvaccess/nvda#14482]
### Steps to reproduce:

1. Run black on a Python project.
2. Let NVDA read the emojis.

### Actual behavior:

Instead of c:\users\quin, I get c:\, users\quin, as two different strings. Everything new that NVDA speaks in the terminal gets split by however many unicode characters were read until you clear the screen. In this case its 3, because Black showed a sparkles emoji, a shortcake emoji, and then a sparkles emoji again.

### Expected behavior:

NVDA shouldn't care about getting unicode characters in its terminal output, and just continue as normal.

#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2022.3.3
#### Windows version:
Windows 11 21H2 (also reproducible on Windows 10).
#### Name and version of other software in use when reproducing the issue:
N/a (it happens with all terminal applications, including Edbrowse in WSL
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
I remember seeing this issue for a while now, but I never paid attention to exact versions. Its not a recent regression though, at least.
#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes

**Comment by josephsl** [comment]
Hi, does changing the diffing algorithm have any impact? Thanks.

**Comment by TheQuinbox** [comment]
Hi,
It doesn't seem to, although odddly enough, the issue seems to have gone away on Windows 11 22H2, with the latest version of Windows Terminal

**Comment by seanbudd** [comment]
If Microsoft has fixed this issue, we can close it
